src/share/classes/com/sun/javatest/regtest/exec/Agent.java

```diff
@@ -90,6 +92,18 @@ public static class Fault extends Exception {
     static final boolean showAgent = Flags.get("showAgent");
     static final boolean traceAgent = Flags.get("traceAgent");
 
+    private static final long UNKNOWN_PID = -1;
+    private static final Method PID_METHOD;
+    static {
+        Method pidMethod = null;
+        try {
+            pidMethod = Process.class.getDeclaredMethod("pid"); // only available in Java 9+
+        } catch (NoSuchMethodException | SecurityException e) {
```

Comment:
It can throw `UnsupportedOperationException`
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Process.html#pid()

Reply:
Good catch. Given that the reason I introduced the pid() call was for better debuggability, I have now updated to PR to catch the broad `Exception` type, both at the place where we reflectively check for the presence of `pid` method as well as where we reflectively call the `pid()` method, to ignore any exceptions and return `-1` in such cases.